spaghetto/src/Spago/Config.purs

```diff
@@ -290,12 +292,43 @@ type WorkspacePackage =
   , hasTests :: Boolean
   }
 
+workspacePackageCodec :: JsonCodec WorkspacePackage
+workspacePackageCodec = CAR.object "WorkspacePackage"
+  { path: CA.string
+  , package: packageConfigCodec
+  , doc: yamlDocCodec
+  , hasTests: CA.boolean
+  }
+
+yamlDocCodec :: JsonCodec (YamlDoc Config)
+yamlDocCodec = CA.codec' decode encode
+  where
+  -- TODO: implementation of encode
+  encode _x = CA.encode CA.null unit
+
+  -- TODO: implementation of decode if needed
+  decode _json = Left MissingValue
+
 data Package
   = RegistryVersion Version
   | GitPackage GitPackage
   | LocalPackage LocalPackage
   | WorkspacePackage WorkspacePackage
 
+packageCodec :: JsonCodec Package
+packageCodec = CA.codec' decode encode
+  where
+  encode (RegistryVersion x) = CA.encode Version.codec x
+  encode (GitPackage x) = CA.encode gitPackageCodec x
+  encode (LocalPackage x) = CA.encode localPackageCodec x
+  encode (WorkspacePackage x) = CA.encode workspacePackageCodec x
```

Comment:
A general note first: the main advantage of using codecs is that we can have different JSON encodings for the same type - we are defining this one (and other related ones) specifically for the `ls` command, so we should have all the new codecs in that module, so that we are free to have different implementations here (if we ever want a generic codec for the config)

Specifically about this codec: in my previous review I suggested that we "just dump the json using the existing codecs", but after trying it out I feel that we should be a little more structured: if you're parsing this data from any language (or even just `jq`), you'll need a way to discern the type of the inner object.
This is commonly done by wrapping the cases of the sum into records with `type` and `value` keys.
So in our case we'd have (I'm putting here just the encode as an example):
```suggestion
packageCodec = CAR.object "Package" { type: CA.string, value: innerCodec }
  where
  innerCodec = CA.codec' decode encode
  
  registryVersionCodec = CAR.object "RegistryVersion" { version: Version.codec }
  
  encode (RegistryVersion x) = CA.encode registryVersionCodec x
  encode (GitPackage x) = CA.encode gitPackageCodec x
  encode (LocalPackage x) = CA.encode localPackageCodec x
  encode (WorkspacePackage x) = CA.encode workspacePackageCodec x
```

Note that I wrapped the `RegistryVersion` one so that all the `value`s are records (for consistency). 
I would also suggest inlining `workspacePackageCodec` and other codecs that we need inside this `where` clause, since they will only be used here.